internal/observability/middleware_test.go

```diff
@@ -386,5 +388,31 @@ func TestMiddlewareStats(t *testing.T) {
 	expected, err := ioutil.ReadFile("testdata/prom.txt")
 	assert.NoError(t, err, "Unexpected error reading testdata.")
 
-	pallytest.AssertPrometheus(t, reg, strings.TrimSpace(string(expected)))
+	assertPrometheus(t, root, strings.TrimSpace(string(expected)))
```

Comment:
Rather than dropping down to assertions using the Prometheus scrape text, let's use the metrics-level test utilities. Since we're only testing with a single request, it should be fairly simple:

```
snap := root.Snapshot()
expected := ... // build expected snapshot here
assert.Equal(t, expected, snap, "Unexpected metrics after end-to-end test.")
```

This should let us delete the scrape utilities below and the testdata directory.